src/Runner/Hook.php

```diff
@@ -98,10 +99,28 @@ public function run(): void
             $this->io->write(['', '<info>No actions to execute</info>'], true, IO::VERBOSE);
             return;
         }
+
         $this->beforeHook();
+
+        $failedActions = 0;
         foreach ($actions as $action) {
-            $this->handleAction($action);
```

Comment:
This nesting is a bit deep. I would propose two methods like `executeAndThrowAtFirstError` and `executeAndThrowAfterAllActions` I'm not so sure about the naming yet, but something easy to understand.

Reply:
I like `executeFailOnFirstError` `executeFailAfterAllAction` ;)